arcgis-ios-sdk-samples/Route and directions/Navigate route with rerouting/NavigateRouteWithReroutingViewController.swift

```diff
@@ -0,0 +1,341 @@
+    /// The destination location, the Fleet Science Center.
+    let destinationLocation = AGSPoint(x: -117.146679, y: 32.730351, spatialReference: .wgs84())
+    /// The location data source provided by a local GPX file.
+    var gpxDataSource: AGSGPXLocationDataSource?
+    
+    /// A formatter to format a time value into human readable string.
+    let timeFormatter: DateComponentsFormatter = {
+        let formatter = DateComponentsFormatter()
+        formatter.allowedUnits = [.hour, .minute, .second]
+        formatter.unitsStyle = .full
+        return formatter
+    }()
+    /// An AVSpeechSynthesizer for text to speech.
+    let speechSynthesizer = AVSpeechSynthesizer()
+    
+    // MARK: Actions
+    
+    /// Called in response to the "Navigate" button being tapped.
+    @IBAction func startNavigation() {
+        navigateBarButtonItem.isEnabled = false
+        resetBarButtonItem.isEnabled = true
+        // Start the location data source and location display.
+        mapView.locationDisplay.start()
+    }
+    
+    /// Called in response to the "Reset" button being tapped.
+    @IBAction func reset() {
+        // Stop the speech, if there is any.
+        speechSynthesizer.stopSpeaking(at: .immediate)
+        // Set the initial location.
+        guard let initialLocation = gpxDataSource?.locations?.first else { return }
+        // Reset to the starting location for location display.
+        mapView.locationDisplay.dataSource.didUpdate(initialLocation)
+        // Stop the location display as well as datasource generation, if reset before the end is reached.
+        mapView.locationDisplay.stop()
+        mapView.locationDisplay.autoPanModeChangedHandler = nil
+        mapView.locationDisplay.autoPanMode = .off
+        directionManeuvers.removeAll()
+        setStatus(message: "Directions are shown here.")
+        
+        // Reset the navigation.
+        setNavigation(with: routeResult)
+        // Reset buttons state.
+        resetBarButtonItem.isEnabled = false
+        navigateBarButtonItem.isEnabled = true
+    }
+    
+    /// Called in response to the "Recenter" button being tapped.
+    @IBAction func recenter() {
+        mapView.locationDisplay.autoPanMode = .navigation
+        recenterBarButtonItem.isEnabled = false
+        mapView.locationDisplay.autoPanModeChangedHandler = { [weak self] _ in
+            DispatchQueue.main.async {
+                guard let self = self else { return }
+                self.recenterBarButtonItem.isEnabled = true
+                self.mapView.locationDisplay.autoPanModeChangedHandler = nil
+            }
+        }
+    }
+    
+    // MARK: Instance methods
+    
+    /// A wrapper function for operations after the route is solved by an `AGSRouteTask`.
+    ///
+    /// - Parameter routeResult: The result of the solve route operation.
+    func didSolveRoute(with routeResult: Result<AGSRouteResult, Error>) {
+        switch routeResult {
```

Comment:
The parameter is a result with a rout result. That's not confusing at all, is it?
```suggestion
    /// - Parameter result: The result of the solve route operation.
    func didSolveRoute(with result: Result<AGSRouteResult, Error>) {
        switch result {
```